File: Sources/MutsumiPet/Models/WindowLayerMode.swift
import AppKit
import CoreGraphics

enum WindowLayerMode: String, CaseIterable, Identifiable {
    case front
    case normal
    case desktop

    var id: String { rawValue }

    var title: String {
        switch self {
        case .front: "始终置顶"
        case .normal: "普通窗口"
        case .desktop: "位于桌面后台"
        }
    }

    var windowLevel: NSWindow.Level {
        switch self {
        case .front:
            .floating
        case .normal:
            .normal
        case .desktop:
            NSWindow.Level(rawValue: Int(CGWindowLevelForKey(.desktopIconWindow)) + 1)
        }
    }
}

File: windows/MutsumiPet.Windows/Models/WindowLayerMode.cs
namespace MutsumiPet.Models
{
    public enum WindowLayerMode
    {
        Front,
        Normal,
        Desktop
    }

    public static class WindowLayerModes
    {
        public static readonly WindowLayerMode[] AllCases =
        {
            WindowLayerMode.Front,
            WindowLayerMode.Normal,
            WindowLayerMode.Desktop
        };

        public static string Title(WindowLayerMode mode)
        {
            switch (mode)
            {
                case WindowLayerMode.Normal: return "普通窗口";
                case WindowLayerMode.Desktop: return "位于桌面后台";
                default: return "始终置顶";
            }
        }

        /// Stable string used for persistence, matching the macOS build's raw values.
        public static string RawValue(WindowLayerMode mode)
        {
            switch (mode)
            {
                case WindowLayerMode.Normal: return "normal";
                case WindowLayerMode.Desktop: return "desktop";
                default: return "front";
            }
        }

        public static WindowLayerMode Parse(string rawValue, WindowLayerMode fallback)
        {
            switch (rawValue)
            {
                case "front": return WindowLayerMode.Front;
                case "normal": return WindowLayerMode.Normal;
                case "desktop": return WindowLayerMode.Desktop;
                default: return fallback;
            }
        }
    }
}
